WolvenKit.RED4/Types/ClassesExt/entEntityTemplate.cs

```diff
@@ -7,15 +7,22 @@ public partial class entEntityTemplate
     [REDProperty(IsIgnored = true)]
     public CHandle<entEntity> Entity
     {
-        get => GetPropertyValue<CHandle<entEntity>>();
+        get => GetPropertyValue<CHandle<entEntity>>()!; // set in PostConstruct, so not nullable
         set => SetPropertyValue<CHandle<entEntity>>(value);
     }
 
     [RED("components")]
     [REDProperty(IsIgnored = true)]
     public CArray<entIComponent> Components
     {
-        get => GetPropertyValue<CArray<entIComponent>>();
+        get => GetPropertyValue<CArray<entIComponent>>()!; // set in PostConstruct, so not nullable
         set => SetPropertyValue<CArray<entIComponent>>(value);
     }
+
+
+    partial void PostConstruct()
```

Comment:
Same here, changes serialization logic

Reply:
but enforces type safety. Entity shouldn't be null, nor should components - not even if serializing. Pretty sure the serializer assigns empty objects there if they are?